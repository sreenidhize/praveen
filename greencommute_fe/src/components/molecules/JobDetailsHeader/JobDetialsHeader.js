import JobCardHeader from "../JobCardHeader/JobCardHeader";
import React from "react";
import { Box, Grid, makeStyles } from "@material-ui/core";
import Button from "../../atom/button/Button";
import PropTypes from "prop-types";

const useStyle = makeStyles({
  root: {
    display: "flex",
  },
  gridItem: {
    flexDirection: "column",
  },
});
function JobDetialsHeader(props) {
  const {
    headerProps,
    button1Label,
    button2Label,
    handleOnSave,
    onClick2,
  } = props;
  const { src, alt, jobTitle, companyName, location } = headerProps;

  return (
    <Grid container direction="column" spacing={4}>
      <Grid item xs={12}>
        <JobCardHeader
          src={src}
          alt={alt}
          jobTitle={jobTitle}
          companyName={companyName}
          location={location}
        />
      </Grid>
      <Grid item container xs={12}>
        <Grid item xs={12} sm={3}>
          <div style={{ display: "flex" }}>
            <Box marginLeft="18%" marginRight="15px">
              <Button
                label={button1Label}
                color="primary"
                type="small"
                variant="outlined"
                onClick={() => handleOnSave()}
              />
            </Box>
            <Box marginLeft="4.5%" marginRight="10px">
              <Button
                label={button2Label}
                color="primary"
                type="small"
                variant="contained"
                onClick={onClick2}
              />
            </Box>
          </div>
        </Grid>
      </Grid>
    </Grid>
  );
}

JobDetialsHeader.propTypes = {
  headerProps: PropTypes.object,
  button1Label: PropTypes.string,
  button2Label: PropTypes.string,
  onClick1: PropTypes.func,
  onClick2: PropTypes.func,
};

JobDetialsHeader.defaultProps = {
  button1Label: "Save",
  button2Label: "Apply",
};

export default JobDetialsHeader;
