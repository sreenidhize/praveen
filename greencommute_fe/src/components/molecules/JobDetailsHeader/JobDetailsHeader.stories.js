import React from "react";
import JobDetialsHeader from "./JobDetialsHeader";

const JobDetailsHeaderStory = {
  title: "Molecules/DetailsHeader",
  component: JobDetialsHeader,
};

const Template = (args) => <JobDetialsHeader {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  headerProps: {
    src: "https://upload.wikimedia.org/wikipedia/commons/4/44/BMW.svg",
    alt: "BMW company",
    jobTitle: "User Experience Designer",
    companyName: "BMW",
    location: "Hi-Tech City, Telangana",
  },
};

export default JobDetailsHeaderStory;
