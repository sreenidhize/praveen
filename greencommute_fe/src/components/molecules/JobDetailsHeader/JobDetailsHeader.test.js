import { render, screen } from "@testing-library/react";
import React from "react";
import JobDetailsHeader from "./JobDetialsHeader";
import userEvent from "@testing-library/user-event";

describe("JobDetailsHeader renders correctly", () => {
  const headerProps = {
    src: "https://upload.wikimedia.org/wikipedia/commons/4/44/BMW.svg",
    alt: "BMW company",
    jobTitle: "User Experience Designer",
    companyName: "BMW",
    location: "Hi-Tech City, Telangana",
  };
  test("should match snapshot", () => {
    const { asFragment } = render(
      <JobDetailsHeader headerProps={headerProps} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("JobDetailsHeader should render logo and buttons should have onClick", () => {
    const onClick1 = jest.fn();
    const onClick2 = jest.fn();
    const { container } = render(
      <JobDetailsHeader
        headerProps={headerProps}
        handleOnSave={onClick1}
        onClick2={onClick2}
      />
    );
    expect(container.getElementsByTagName("img").length).toBe(1);
    expect(container.getElementsByTagName("h4").length).toBe(3);
    expect(container.getElementsByTagName("button").length).toBe(2);
    userEvent.click(screen.getByText("Save"));
    expect(onClick1).toHaveBeenCalled();
    userEvent.click(screen.getByText("Apply"));
    expect(onClick2).toHaveBeenCalled();
  });
});
