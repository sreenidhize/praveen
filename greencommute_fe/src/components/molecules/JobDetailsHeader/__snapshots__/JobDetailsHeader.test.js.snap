// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDetailsHeader renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root makeStyles-root-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-smallMedia-4"
          >
            <img
              alt="BMW company"
              class="MuiAvatar-img"
              src="https://upload.wikimedia.org/wikipedia/commons/4/44/BMW.svg"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-columnMargin-2 MuiGrid-container MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h4
              class="MuiTypography-root makeStyles-primary-5 MuiTypography-h4"
            >
              User Experience Designer
            </h4>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h4
              class="MuiTypography-root makeStyles-secondary-6 MuiTypography-h4"
            >
              BMW
            </h4>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h4
              class="MuiTypography-root makeStyles-secondary-6 MuiTypography-h4"
            >
              Hi-Tech City, Telangana
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          style="display: flex;"
        >
          <div
            class="MuiBox-root MuiBox-root-11"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-base-12 makeStyles-small-13 MuiButton-outlinedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Save
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiBox-root MuiBox-root-18"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-base-12 makeStyles-small-13 MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Apply
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
